### PR TITLE
【Infer Symbolic Shape BUAA No.206】Add overlap_add op

### PR DESCRIPTION
**Pull request by lwkhahaha**
### PR Category

User Experience

### PR Types

Others

### Description

Add overlap_add op

### CODE REVIEW

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc [hunk 1]
```diff
@@ -2046,12 +2046,89 @@ bool OneHotOpInferSymbolicShape(pir::Operation *op,
...
+  std::vector<symbol::DimExpr> output_shape;
+  symbol::DimExpr n_frames;
+  int frame_length;
+  symbol::DimExpr seq_length;
+
+  int start_axis = 0;
+  int end_axis = 0;
+  if (axis == 0) {
+    n_frames = x_dims[0];
+    frame_length = static_cast<int64_t>(x_dims[0].Get<std::int64_t>());
```
Comment (gongshaotian) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc]: x_dims[0]不一定能cast成int

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc [hunk 2]
```diff
@@ -2046,12 +2046,89 @@ bool OneHotOpInferSymbolicShape(pir::Operation *op,
...
+    start_axis = 2;
+    end_axis = x_rank - 1;
+  } else {
+    n_frames = x_dims[x_rank - 1];
+    frame_length = static_cast<int64_t>(x_dims[x_rank - 2].Get<std::int64_t>());
```
Comment (gongshaotian) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc]: 同上

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc [hunk 3]
```diff
@@ -2046,12 +2046,89 @@ bool OneHotOpInferSymbolicShape(pir::Operation *op,
...
+    start_axis = 0;
+    end_axis = x_rank - 3;
+  }
+
+  PADDLE_ENFORCE_LE(
```
Comment (gongshaotian) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/unary_infer_sym.cc]: 这个enforce直接不要了，是不是就不需要把frame_length cast成int类型了

**Comment by gongshaotian** [comment]
https://github.com/PaddlePaddle/Paddle/pull/67692

**Review by gongshaotian — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]

**Pull request merged**